Native profiler v0.13.0 (#18883)
DataDog/documentation: content/en/profiler/profiler_troubleshooting/ddprof.md

## Missing profiles in the profile search page

> If you've configured the profiler and don't see profiles in the profile search page, turn on verbose logging (`-l debug`) and [open a support ticket][1] with log files and the following information:

Reviewer: ```suggestion
If you've configured the profiler and don't see profiles in the profile search page, turn on verbose logging (`-l debug`) and [open a support ticket][1]. In the support ticket, include log files along with the following information:
```

- Linux kernel version (`uname -r`)
- libc version (`ldd --version`)
- Value of `/proc/sys/kernel/perf_event_paranoid`
- Complete command line, including both profiler and application arguments

> The section bellow lists potential setup issues.

Reviewer 2: ```suggestion
The section below lists potential setup issues.
```

> - `CAP_PERFMON`: adds BPF and `perf_event_open` capabilities (available on Linux v5.8 or later)

Reviewer: ```suggestion
- `CAP_PERFMON`: Adds BPF and `perf_event_open` capabilities (available on Linux v5.8 or later).
```

Reviewer: I have the same feedback for the line above this one for CAP_SYS_ADMIN (adding a period and capitalizing the first letter of the word "Add" in the description), but I'm unable to add it as a suggested change since it's outside the scope of the PR (i.e., Github won't let me do it).

There are a few less common permissions issues:
- The profiler relies upon the `perf_event_open()` syscall, which is disallowed by some container runtimes. Check the appropriate documentation to see whether this might be the case.
- Some seccomp profiles forbid `perf_event_open()`. If your system runs such a configuration, you may not be able to run the profiler.

### "\<ERROR\> Could not mmap memory for watcher

> The profiler uses pinned memory to store (performance) events. This type of memory is constrained by kernel settings. You can view your current setting using `ulimit -l`. The following capability can be used to bypass this limitation:

Reviewer 2: Suggestion to change the qualification.  Rather than describing how the product operates, describe the rationale (it's necessary).
```suggestion
The profiler requires pinned memory to operate. This type of memory is constrained by kernel settings. You can view your current setting using `ulimit -l`. The following capability can be used to bypass this limitation:
```

> - `CAP_IPC_LOCK`: allows the use of locked memory (memory exempt from paging)

Reviewer: ```suggestion
- `CAP_IPC_LOCK`: Allows the use of locked memory (memory exempt from paging).
```

### "\<WARNING\> Failure to establish connection"

> This error usually means that the profiler is unable to connect to the Datadog Agent. Enable configuration logging(`--show_config`) to identify the hostname and port number used by the profiler for uploads. Additionally, the content of the error message may relay the hostname and port used. Compare these values to your Agent configuration. Check the profiler's help section (`ddprof --help`) for further information on how to configure the agent's URL.

Reviewer: ```suggestion
This error usually means that the profiler is unable to connect to the Datadog Agent. Enable configuration logging (`--show_config`) to identify the hostname and port number used by the profiler for uploads. Additionally, the content of the error message may relay the hostname and port used. Compare these values to your Agent configuration. Check the profiler's help section (`ddprof --help`) for further information on how to configure the Agent's URL.
```

## Profiles are empty or sparse

The root of your profile is the frame annotated with the application's binary.  If this frame shows a significant amount of CPU time, but no child frames, consider the following:
- Stripped binaries do not have symbols available. Try using a non-stripped binary or a non-minified container image.
- Certain applications and libraries benefit from their debug packages being installed. This is true for services installed through your repo's package manager or similar.

> If you are seeing anonymous frames, you might be using a interpreted or JITed language. Consider enabling perf maps or JIT dump information.

Reviewer 2: Should we describe what an anonymous frame looks like?
